src/server/cmd/worker/main.go

```diff
@@ -86,9 +86,19 @@ func do(config interface{}) error {
 	}
 
 	// keepalive forever
-	if _, err := env.GetEtcdClient().KeepAlive(context.Background(), resp.ID); err != nil {
+	keepAliveChan, err := env.GetEtcdClient().KeepAlive(context.Background(), resp.ID)
+	if err != nil {
 		return errors.Wrapf(err, "error with KeepAlive")
 	}
+	go func() {
+		for {
+			_, more := <-keepAliveChan
+			if !more {
+				log.Errorf("failed to renew worker ip etcd lease")
```

Comment:
From my understanding, if there's any actual problem renewing the lease, some actual operation will fail later, right? Obviously this doesn't introduce any new bug here.

Reply:
I think it is mostly used for collecting info from workers, although the restart datum functionality would be affected. We should eventually move to using kubernetes for this rather than having the workers register their ip in etcd.